tensorboard/schema.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TensorBoard Database Schemas.
+
+This module adds classes to define the database scheme for TensorBoard.
+The classes provide a database independent way to specify the scheme
+so that we can easily add support for new databases.
+
+WARNING: This module is EXPERIMENTAL. It will not be considered stable
+until data migration tools are put into place. Until that time, any
+database created with this schema will need to be deleted as new updates
+occur.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+class ColumnType(object):
+  pass
+
+
+class BoolColumnType(ColumnType):
+  pass
+
+
+class BytesColumnType(ColumnType):
+  def __init__(self, length=None):
+    """Define a bytes column.
+
+    Args:
+      length: The length of the string. None indicates column
+        should have maximum length allowed.
+
+    : type lenth: int | None
+    : rtype: BytesColumnType
+    """
+    self.length = length
+
+
+class Int64ColumnType(ColumnType):
+  pass
+
+
+class StringColumnType(ColumnType):
+
+  def __init__(self, length=None):
+    """Define a string column.
+
+    Args:
+      length: The length of the string. None indicates column
+        should have maximum length allowed.
+
+    : type lenth: int | None
+    : rtype: StringColumnType
+    """
+    self.length = length
+
+# TODO(jlewi): Add a description field to columns?
+
+
+class ColumnSchema(object):
+  """Defines the schema for a column."""
+
+  def __init__(self, name, value_type, not_null=False):
+    """Define a column.
+
+    Args:
+      name: Name of the column.
+      value_type: A ColumnType object describing the type for the
+        column.
+      not_null: If true then column is required to be not null.
+
+    : type name: str
+    : type value_type: list[ColumnType]
+    : type not_null: list[bool]
+    : rtype: ColumnSchema
+    """
+    self.name = name
+    self.value_type = value_type
+    self.not_null = not_null
+
+
+class TableSchema(object):
```

Comment:
Keep in mind that plugin authors referencing this file to create their own tables, would have code that ends up looking like this: `schema.TableSchema(...)`. So I'd be fine if this was just named `Table`.

Reply:
Reverted.